core/embed/extmod/modtrezormintlayer/modtrezormintlayer.h

```diff
@@ -0,0 +1,693 @@
+    return;
+  }
+
+  switch (res->len_or_err.err) {
+    case WrongHashSize:
+      mp_raise_ValueError("Invalid hash size");
+      break;
+    case InvalidUtxoType:
+      mp_raise_ValueError("Invalid UTXO type");
+      break;
+    case InvalidAmount:
+      mp_raise_ValueError("Invalid amount");
+      break;
+    case InvalidAccountCommand:
+      mp_raise_ValueError("Invalid account command");
+      break;
+    case InvalidDestination:
+      mp_raise_ValueError("Invalid destination");
+      break;
+    case InvalidIsTokenUnfreezable:
+      mp_raise_ValueError("Invalid token unfreezable flag");
+      break;
+    case InvalidIsTokenFreezable:
+      mp_raise_ValueError("Invalid token freezable flag");
+      break;
+    case InvalidVrfPublicKey:
+      mp_raise_ValueError("Invalid VRF public key");
+      break;
+    case InvalidPublicKey:
+      mp_raise_ValueError("Invalid public key");
+      break;
+    case InvalidOutputTimeLock:
+      mp_raise_ValueError("Invalid output time lock");
+      break;
+    case InvalidTokenTotalSupply:
+      mp_raise_ValueError("Invalid token total supply");
+      break;
+    default:
+      mp_raise_ValueError("Unknown error");
+      break;
+  }
+}
+
+/// def encode_utxo_input(tx_hash: bytes, index: int, utxo_type: int) -> bytes:
+///     """
+///     encodes an utxo input from tx_hash and index
+///     """
+STATIC mp_obj_t mod_trezormintlayer_utils_mintlayer_encode_utxo_input(
+    mp_obj_t tx_hash, mp_obj_t index, mp_obj_t utxo_type_obj) {
+  mp_buffer_info_t hash = {0};
+  mp_get_buffer_raise(tx_hash, &hash, MP_BUFFER_READ);
+  if (hash.len != 32) {
+    printf("invalid hash len: %ld", (long int)hash.len);
+    mp_raise_ValueError("Invalid hash");
+  }
+  uint32_t idx = trezor_obj_get_uint(index);
+  uint32_t utxo_type = trezor_obj_get_uint(utxo_type_obj);
+  ByteArray arr =
+      mintlayer_encode_utxo_input(hash.buf, hash.len, idx, utxo_type);
+  handle_err(&arr);
+
+  vstr_t pkh = {0};
+  vstr_init_len(&pkh, arr.len_or_err.len);
+  int i = 0;
+  for (; i < arr.len_or_err.len; i++) {
+    ((uint8_t *)pkh.buf)[i] = (uint8_t)arr.data[i];
+  }
+
+  return mp_obj_new_str_from_vstr(&mp_type_bytes, &pkh);
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(
+    mod_trezormintlayer_utils_mintlayer_encode_utxo_input_obj,
+    mod_trezormintlayer_utils_mintlayer_encode_utxo_input);
+
+/// def encode_account_spending_input(nonce: int, delegation_id: bytes, amount:
+/// bytes) -> bytes:
+///     """
+///     encodes an utxo account spendinf from nonce and delegation id
+///     """
+STATIC mp_obj_t
+mod_trezormintlayer_utils_mintlayer_encode_account_spending_input(
+    mp_obj_t nonce_obj, mp_obj_t delegation_id, mp_obj_t amount_obj) {
+  uint64_t nonce = trezor_obj_get_uint64(nonce_obj);
+
+  mp_buffer_info_t hash = {0};
+  mp_get_buffer_raise(delegation_id, &hash, MP_BUFFER_READ);
+  if (hash.len != 32) {
+    printf("invalid hash len: %ld", (long int)hash.len);
+    mp_raise_ValueError("Invalid hash");
+  }
```

Comment:
Let's not repeat boilerplate over and over. E.g. this can be moved into a function `mp_get_hash_raise`, which would call `mp_get_buffer_raise` and then check the length.